### PR TITLE
Updating workflow and fixing warnings

### PR DESCRIPTION
**Pull request by divekarshubham**
*Issue #, if available:*

*Description of changes:*
* Updated CI workflow to add `-Wall -Wextra -Werror` flags
* The buffers `pOtaAgentStateStrings` and `pOtaEventStrings` are guarded by checks to see if loggings is enabled. This is because all the usages of these buffers are in log strings.
* Type casts to to compatible data
* Suppressed unused parameters using `( void )` cast

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/ota.c [hunk 1]
```diff
@@ -305,42 +305,47 @@ static OtaStateTableEntry_t otaTransitionTable[] =
     { OtaAgentStateAll,                 OtaAgentEventShutdown,            shutdownHandler,        OtaAgentStateStopped             },
 };
 
+#ifndef LogInfo
```
Comment (leegeth) [inline source/ota.c:308]: Did you mean ifdef? and why LogInfo?

File: source/ota.c [hunk 2]
```diff
@@ -305,6 +305,8 @@ static OtaStateTableEntry_t otaTransitionTable[] =
     { OtaAgentStateAll,                 OtaAgentEventShutdown,            shutdownHandler,        OtaAgentStateStopped             },
 };
 
+
+
```
Comment (leegeth) [inline source/ota.c:309]: Nit: These extra lines can be removed

File: source/ota.c [hunk 3]
```diff
@@ -324,6 +326,8 @@ static const char * pOtaAgentStateStrings[ OtaAgentStateAll + 1 ] =
     "All"
 };
 
+
+
```
Comment (leegeth) [inline source/ota.c:330]: Nit: These extra lines can be removed

File: source/ota.c (hunk 1 above)
Comment (divekarshubham) [reply source/ota.c]: Thanks for the callout, suppressed these warnings in OTA_Init()

**Review by leegeth — APPROVED** [review]

**Review by leegeth — DISMISSED** [review]

File: source/ota.c (hunk 3 above)
Comment (divekarshubham) [reply source/ota.c]: Done: https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/112/commits/130a2cafd84f8880cf198de5ac2fe2542fdcd831

File: source/ota.c (hunk 2 above)
Comment (divekarshubham) [reply source/ota.c]: Done: https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/112/commits/130a2cafd84f8880cf198de5ac2fe2542fdcd831

**Review by cobusve — DISMISSED** [review]

**Review by pvyawaha — APPROVED** [review]

**Review by abhidixi11 — APPROVED** [review]

**Pull request merged**